src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java

```diff
@@ -106,6 +107,12 @@ protected NumberFormat initialValue() {
         }
     };
 
+    private static final Pattern invalidAspectChar = Pattern.compile("[^.0-9A-Za-z_]");
+
+    private static String sanitiseAspect(String aspect) {
+        return invalidAspectChar.matcher(aspect).replaceAll("_");
```

Comment:
Isn't that doing the opposite of what we want to do? like replacing all valid characters with underscore?

Reply:
Notice the character set in the regular expression: first one is a ^, so it matches everything not in the following list.

Reply:
I indeed missed that. Thanks!